### PR TITLE
Arthig/logging changes 1

### PR DESCRIPTION
**Pull request by agokarn**
ohttp-server logs now appear in kusto
 1 RawMessage
 2 {"log":"{\"timestamp\":\"2024-10-08T20:59:53.358554Z\",\"level\":\"INFO\",\"fields\":{\"message\":\"Encapsulated final chunk 00465f...9027f4(0,72)\"},\"target\":\"ohttp\"}","stream":"stdout","time":"2024-10-08T20:59:53.358663364Z"}
 3 {"log":"{\"timestamp\":\"2024-10-08T20:59:53.358522Z\",\"level\":\"INFO\",\"fields\":{\"message\":\"Response nonce f2607a...f20a92(32)\"},\"target\":\"ohttp\"}","stream":"stdout","time":"2024-10-08T20:59:53.358659964Z"}
 4 {"log":"{\"timestamp\":\"2024-10-08T20:59:53.358518Z\",\"level\":\"INFO\",\"fields\":{\"message\":\"openai-processing-ms: 233\"},\"target\":\"ohttp_server\"}","stream":"stdout","time":"2024-10-08T20:59:53.358656563Z"}

### CODE REVIEW

File: ohttp-client/src/main.rs [hunk 1]
```diff
@@ -225,14 +225,23 @@ async fn main() -> Res<()> {
...
+    trace!("================== STEP 2 ==================");
 
     let (enc_request, client_response) = ohttp_request.encapsulate(&request_buf)?;
-    info!(
-        "Sending encrypted OHTTP request to {}: {}",
-        args.url,
-        hex::encode(&enc_request[0..60])
-    );
+    let enc_request_len = enc_request.len();
+    if enc_request_len > 6 {
```
Comment (kapilvgit) [inline ohttp-client/src/main.rs:232]: I am not sure if the first few bytes will help debugging. I recommend we condition on a flag in the request, and dump the entire encrypted request if the flag is set.

File: ohttp-server/src/main.rs [hunk 2]
```diff
@@ -91,7 +92,8 @@ async fn import_config(maa: &str, kms: &str) -> Res<KeyConfig> {
         panic!("Failed to get MAA token. You must be root to access TPM.")
     };
     let token = String::from_utf8(tok).unwrap();
-    info!("Fetched MAA token: {token}");
+    info!("Fetched MAA token");
+    trace!("{token}");
```
Comment (kapilvgit) [inline ohttp-server/src/main.rs:96]: I would do this only conditioned on the client's consent.

File: ohttp-server/src/main.rs [hunk 3]
```diff
@@ -129,10 +131,8 @@ async fn import_config(maa: &str, kms: &str) -> Res<KeyConfig> {
             let skr: ExportedKey =
                 from_str(&skr_body).expect("Failed to deserialize SKR response. Check KMS version");
 
-            info!(
-                "SKR successful, KID={}, Receipt={}, Key={}",
-                skr.kid, skr.receipt, skr.key
-            );
+            info!("SKR successful");
+            trace!("KID={}, Receipt={}, Key={}", skr.kid, skr.receipt, skr.key);
```
Comment (kapilvgit) [inline ohttp-server/src/main.rs:135]: I would not dump the key in any scenario.

File: verifier/src/lib.rs [hunk 4]
```diff
@@ -58,7 +58,7 @@ fn compute_leaf(leaf_components: LeafComponents) -> Res<Vec<u8>> {
     hasher.update(leaf_components.commit_evidence.as_bytes())?;
     let mut commit_evidence_digest = hasher.finish()?.to_vec();
 
-    info!(
+    /* info!(
```
Comment (kapilvgit) [inline verifier/src/lib.rs:61]: Why are we disabling this tracing on the client?

File: ohttp-server/src/main.rs (hunk 2 above)
Comment (agokarn) [reply ohttp-server/src/main.rs:96]: looks like IGVM agent does this too. Also there is a response header "x-ms-request-id" that gets populated if we need to get help from MAA debugging failures

File: ohttp/src/lib.rs [hunk 5]
```diff
@@ -314,12 +314,21 @@ impl ServerResponse {
...
-        let response_nonce = Ok(self.response_nonce.clone());
-        info!(
-            "Response nonce {}",
-            hex::encode(self.response_nonce.clone())
-        );
-        let nonce_stream = once(async { response_nonce });
+        let response_nonce_vec: Vec<u8> = self.response_nonce.to_vec();
+        let result = Ok(response_nonce_vec.clone());
+        let response_nonce_len = response_nonce_vec.len();
+        if response_nonce_len > 6 {
```
[the remark this answers is not in the document]
Comment (agokarn) [reply ohttp/src/lib.rs]: I agree. This was getting too complicated. So the length is part of info! and the encrypted messages part of trace!

File: ohttp/src/lib.rs [hunk 6]
```diff
@@ -471,15 +476,16 @@ impl ClientResponse {
...
+                trace!("{}", hex::encode(&enc_response));
                 buffer.append(&mut enc_response);
                 info!("Buffer size {}", buffer.len());
 
                 // Response Nonce (Nk)
                 if !nonce_received && buffer.len() >= nonce_size {
                     nonce_received = true;
                     let nonce: Vec<_> = buffer.drain(0..nonce_size).collect();
-                    info!("Setting response nonce: {}({})", hex::encode(&nonce), nonce.len());
+                    info!("Setting response nonce: {}...({})", hex::encode(&nonce[..16]), nonce.len());
```
Comment (kapilvgit) [inline ohttp/src/lib.rs:488]: May be ok to log the entire nonce

File: ohttp/src/lib.rs [hunk 7]
```diff
@@ -316,8 +316,9 @@ impl ServerResponse {
         // Response Nonce (Nk)
         let response_nonce = Ok(self.response_nonce.clone());
         info!(
-            "Response nonce {}",
-            hex::encode(self.response_nonce.clone())
+            "Response nonce {}...({})",
+            hex::encode(&self.response_nonce.clone()[..16]),
```
Comment (kapilvgit) [inline ohttp/src/lib.rs:320]: May be ok to log the entire nonce.

**Review by kapilvgit — APPROVED** [review]

File: ohttp-client/src/main.rs [hunk 8]
```diff
@@ -1,7 +1,7 @@
 use bhttp::{Message, Mode};
 use clap::Parser;
 use futures_util::{stream::unfold, StreamExt};
-use log::info;
+use log::{info, trace};
```
[the remark this answers is not in the document]
Comment (agokarn) [reply ohttp-client/src/main.rs:4]: We could. But I dont see a need to. In the server we need json and the target component,

**Pull request merged**